sdks/java/harness/src/main/java/org/apache/beam/fn/harness/state/MultimapUserState.java

```diff
@@ -65,13 +62,11 @@
   private boolean isClosed;
   private boolean isCleared;
   // Pending updates to persistent storage
-  private HashSet<K> pendingRemoves = Sets.newHashSet();
-  private HashMap<K, List<V>> pendingAdds = Maps.newHashMap();
-  // Map keys with no values in persistent storage
-  private HashSet<K> negativeCache = Sets.newHashSet();
```

Comment:
How come you removed this?

Reply:
Storing the empty iterable in persistedValues can represent the negative cache as well since the underlying PrefetchableIterable caches the first page and hence iterating over it is cheap.

This also ties in with a future change where more of the PrefetchableIterable is cached.